client/internal/debug/debug.go

```
@@ -580,6 +600,40 @@ func (g *BundleGenerator) addSingleLogfile(logPath, targetName string) error {
 	return nil
 }
 
+// addSingleLogFileGz adds a single gzipped log file to the archive
+func (g *BundleGenerator) addSingleLogFileGz(logPath, targetName string) error {
+	f, err := os.Open(logPath)
+	if err != nil {
+		return fmt.Errorf("open gz log file %s: %w", targetName, err)
+	}
+	defer f.Close()
+
+	gzr, err := gzip.NewReader(f)
+	if err != nil {
+		return fmt.Errorf("create gzip reader: %w", err)
+	}
+	defer gzr.Close()
+
+	pr, pw := io.Pipe()
+	go func() {
+		anonymizeLog(gzr, pw, g.anonymizer)
+		pw.Close()
+	}()
+
+	var buf bytes.Buffer
+	gw := gzip.NewWriter(&buf)
+	if _, err := io.Copy(gw, pr); err != nil {
+		return fmt.Errorf("re-gzip: %w", err)
+	}
+	gw.Close()
```

Let's handle the close errors. The writer ones are critical and should return an error